### PR TITLE
Who is in - GUI

### PR DESCRIPTION
**Pull request by user84918342**

### CODE REVIEW

File: src/components/BookingForm.vue [hunk 1]
```diff
@@ -51,14 +51,14 @@
         <v-col>
           <bad-contained-button
             id="btnBook"
-            :click="submitBooking"
+            :click="handleButtonClick"
```
Comment (AlexGagne) [inline src/components/BookingForm.vue]: `handleButtonClick` is a little bit too generic of a name. Generally, you want to name functions so that someone can understand roughly what the function does without having to read the implementation.

In this case, it's not possible here, `handleButtonClick` could be doing literally anything.

`submitBooking` was a good name for it. But since it collides with an existing function name, what about `bookADesk`? You can pick any name here, as long as it describes what the function is doing better than `handleButtonClick`

File: src/components/BookingForm.vue [hunk 2]
```diff
@@ -142,16 +148,31 @@ export default {
...
+      const bookings = await getAsync(`/bookings?date=${this.bookingDate}&office=${this.selectedOffice.id}`)
+
+      // The Email is temporarily used as a username
+      // In the future real usernames and profile pictures will be supported
+      const extractUsername = (booking) => {
+        const [username] = booking.user.email.split("@");
+        return username;
+      };
+
+      this.bookings = bookings.data.items.map(extractUsername)
```
Comment (AlexGagne) [inline src/components/BookingForm.vue]: Generally, when writing code, you should try to follow what's called the "Single Responsibility Principle". Essentially, each class / module / vue should have a single thing that they do. In this case, the logic for choosing what information from a booking should be displayed belongs, in my opinion, in the BadUserListItem.

BookingForm takes care of holding all of the components of the BookingForm. (Note: technically we can say it does too much since it is also responsible for calling the APIs and handling the responses, but I think it's out of scope of your PR). Then the BadUserList's responsibility is to hold the list of BadUserListItem. BadUserListItem then chooses from the Booking what information they want to display and how.

This has the following advantage:
Let's say we add user pictures to the bookings. With my suggested change you would only need to change BadUserListItem to handle the new picture. Instead of now, where you would have to change BookingForm, BadUserList and BadUserListItem

File: src/components/BadUserList.vue [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<template>
```
Comment (AlexGagne) [inline src/components/BadUserList.vue:1]: When you add new code, it is super important to also add new tests (or really, the ideal way of working is to add automated tests first and then write the code, a technique known as Test Driven Development).

The goal with tests is to enforce all of the little details and edge cases so that someone does not need to understand all of the details of your implementation if they want to modify it. They can run your tests and know, if one of them breaks, that they might have accidentally broken the code. For a small program like Book a Desk, this might seem trivial, but Book a Desk will grow. At a certain size, it becomes impossible to know if you broke something unless a test is there to catch it for you.

So here are the tests that I think you should be adding:

`Given a date and a user list When displaying the user list Then the correct date is shown in the heading`

`Given a list of bookings and a user list When displaying the user list Then the users are shown`
( this test is specifically about having 1 item per booking)

`Given a booking When displaying the booking Then the username is shown`
(this test is for the BadUserListItem, where we make sure the username is shown)

File: src/components/BadUserListItem.vue [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        };
+    },
+    computed: {
+      identicon: function() {
+        return jdenticon.toSvg(this.username, this.size);
+      },
+      username: function() {
+        // The Email is temporarily used as a username
+        // In the future real usernames and profile pictures will be supported
+        const extractUsername = (booking) => {
```
Comment (AlexGagne) [inline src/components/BadUserListItem.vue]: Since we are no longer mapping to a list of bookings, I think it is simpler to just have the "extract username" written in-line, without an extract username function

File: src/components/BookingForm.vue [hunk 5]
```diff
@@ -169,6 +182,11 @@ export default {
       this.messageType = "error";
       this.isMessageShownOnBooking = true;
     },
+    async bookADesk() {
```
Comment (AlexGagne) [inline src/components/BookingForm.vue:180]: The code for book a desk has some redundant calls.
bookADesk calls:
1. `submitBooking`, which then:
    - Calls post on /bookings
    - If there are no errors, it calls `fetchAvailabilities` and then calls `fetchBookings`
    - If there is an error, calls `displayErrorMessage`
2. `fetchAvailabilities` (already done)
3. `fetchBookingsList` (is `fetchBookings` or `fetchBookingsList` redundant?)

From this, I think:
1. We should somehow reconcile both `submitBooking` and `bookADesk`, since they both have redundant calls
2. Look into if `fetchBookings` is necessary. I forget exactly why it was used, but I don't know if it's necessary anymore
    - If `fetchBookings` is no longer necessary, please remove it. Also rename `fetchBookingsList` to `fetchBookings`. When naming, you generally want to avoid including type information in the function (such as Array or List). The important part is that the bookings are being fetched, not that they come in a list or not.
    - If `fetchBookings` is necessary, I think we should rename `fetchBookings` and `fetchBookingsList` in a way that describes in what feature they are being used. Right now, they essentially have the same name which makes it very confusing to understand the existence of either function

File: tests/unit/components/BookingForm.spec.js [hunk 6]
```diff
@@ -22,6 +22,22 @@ Vue.component('BadMessage', BadMessage)
 
 const localVue = createLocalVue();
 
+function checkCalledEndpoint(url){
```
Comment (AlexGagne) [inline tests/unit/components/BookingForm.spec.js]: Slight naming convention:
When calling something "check", I expect the function to return a boolean (or maybe doing an assert). Essentially returning has "called endpoint" been checked (which doesn't make much sense as a question).

A better name might be `getCalledEndpoint`

File: src/components/BookingForm.vue (hunk 5 above)
Comment (AlexGagne) [reply src/components/BookingForm.vue:180]: Ok, so I've figured out something about `fetchBookings`. We have a partially implemented UI that handles "Upcoming Bookings" (ie: what bookings the user have that are upcoming". This is displayed in the Dashboard.vue file.

I think the goal was that Dashboard.vue becomes the new main vue instead of BookingForm.vue.

I think there are a couple of things we should discuss in person friday:
1. How to handle both the upcoming bookings and the Who's In Vue.
2. How to refactor Booking Form so that it becomes less of a god class
3. How to use the concept of a store (which is what `fetchBookings` is using). I think this will greatly simplify the code once we properly use the concept.

For now: remove `fetchBookings`, we will reimplement the logic later, after we've done a refactor and when we start to reimplement the "Upcoming Bookings" logic

Comment (AlexGagne) [reply src/components/BookingForm.vue:180]: *also to note: points 1-3 are not required for your PR. We will work on this on a separate PR to make sure each PR does not grow too much and becomes too hard to understand

File: tests/unit/components/BookingForm.spec.js (hunk 6 above)
Comment (user84918342) [reply tests/unit/components/BookingForm.spec.js]: Right, we did the whole function as a group on Friday to let the tests pass (because we now have multiple `GET /bookings` that return different results and axios doesn't seem to have nice way to handle this when mocking)
